### PR TITLE
feat(expermient): Add useExperiment hook

### PR DESCRIPTION
**Pull request by malwilley**
This ports the existing `withExperiment` HoC to a hook. I'll leave in the HoC for now since we still have a significant number of class components, but this hook should be used when possible.

### CODE REVIEW

**Review by scefali — COMMENTED** [review]
@malwilley Should we update the HoC to be functional so it can use this hook? That would avoid having to replicate logic I think.

File: static/app/utils/useExperiment.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+};
+
+function useExperimentAssignment(experiment: ExperimentKey) {
+  const organization = useOrganization();
+  const {user} = useLegacyStore(ConfigStore);
+  const {type} = experimentConfig[experiment];
+
+  if (type === ExperimentType.Organization) {
+    const key = experiment as keyof OrgExperiments;
+    return organization?.experiments?.[key] ?? unassignedValue;
```
Comment (scefali) [inline static/app/utils/useExperiment.tsx:48]: @malwilley Hmmm It seems like we should throw some errors if we have an experiment type as organization/user but the organization/user is falsy

Comment (scefali) [reply static/app/utils/useExperiment.tsx:48]: I realize `useOrganization` is going to blow up if org is falsy as well.

Comment (malwilley) [reply static/app/utils/useExperiment.tsx]: Good thinking! Added some Sentry captureMessage's for when the data doesn't match up correctly.

Do you think there are use cases where we want to use this outside the org context?

Comment (scefali) [reply static/app/utils/useExperiment.tsx]: @malwilley I can't think of any right now. I think it's fine to assume the org context is there.

File: static/app/utils/withExperiment.tsx [hunk 2]
```diff
@@ -99,53 +88,19 @@ function withExperiment<
...
-
-      if (type === ExperimentType.User) {
-        const key = experiment as keyof UserExperiments;
-        return ConfigStore.get('user').experiments[key];
-      }
-
-      return unassignedValue;
-    }
-
-    logExperiment = () =>
```
Comment (scefali) [inline static/app/utils/withExperiment.tsx:132]: @malwilley Nice, luv 2 delete code

**Review by scefali — APPROVED** [review]

**Pull request merged**